### ISSUE TITLE
Sessions don't work

### ISSUE THREAD

**Issue by Ar4ikov** [miguelgrinberg/Flask-SocketIO#698]
Hello there!
My sessions don't wok with socketIO app
```python
from flask import Flask, session, request, jsonify, render_template
from flask_socketio import SocketIO

app = Flask("chat", template_folder="../templates")
app.config["SECRET_KEY"] = "get87di4es8u322#W#Q"

sio = SocketIO(app)

@app.route("/", methods=["GET"])
def main():
    session["user"] = request.remote_addr
    print(session)

    return render_template("/test.html")

@sio.on("entry")
def entry(data):
    print(data.get("user"))
    print(session.get("user"))

sio.run(app)
```

I trying to print session. Then I got
```pytnon-output
Session in app.route
<SecureCookieSession {'user': '127.0.0.1'}>
Hello, World!

Session in sio.on
None
```

what i need to do? how i can fix it?

**Comment by Ar4ikov** [comment]
I also tried to use Redis and set `manage_session=False`, but its not work
```python
# | Created by Ar4ikov
# | Время: 27.04.2018 - 22:09
from flask import Flask, session, request, jsonify, render_template
from flask_socketio import SocketIO
from redis import StrictRedis
from flask_session import Session

app = Flask("chat", template_folder="../templates")
app.config["SECRET_KEY"] = "get87di4es8u322#W#Q"
app.config["SESSION_TYPE"] = "redis"
app.config["SESSION_REDIS"] = StrictRedis(host="10.0.0.2", port=6379, db=0)

Session(app)
sio = SocketIO(app, async_mode="eventlet", manage_session=False)

@app.before_request
def not_perm():
    session.permanent = False

@app.route("/test", methods=["GET"])
def test():
    print(session)

    return render_template("/test2.html", kek=str(session))

@app.route("/", methods=["GET"])
def main():
    session["user"] = request.remote_addr
    # Printing session in Flask app handler
    print(session)

    sio.emit("success", {
        "user": session.get("user")
    })

    return render_template("/test.html")

@sio.

@sio.on("entry")
def entry(data):
    # print(data)
    session.permanent = False
    session["kek"] = False
    # Priniting the session in socektio handler
    print(session)

sio.run(app)
```

Return this

```python-logs

<RedisSession {'_permanent': False, 'user': '127.0.0.1'}>
<RedisSession {'_permanent': False, 'kek': False}>

```

**Comment by miguelgrinberg** [comment]
What's the output of `pip freeze` with your virtual environment activated?

**Comment by miguelgrinberg** [comment]
@Ar4ikov Are you using flask 1.0 by any chance? I just fixed a compatibility issue with sessions in the version of Flask. Please upgrade and try again if that's the version you are using.

**Comment by Ar4ikov** [comment]
@miguelgrinberg Yes, I use Flask 1.0

**Comment by Ar4ikov** [comment]
@miguelgrinberg I successfully have updated just now 
And I had fixed my problem.
Win Firewall hate sessions, and python hadn't had access in it

**Comment by Ar4ikov** [comment]
Its looks like I'm crazy, yeah man

**Comment by Sphinxs** [comment]
I have the same problem, you just updated and it was solved ? `Flask>=1.0.2`

**Comment by miguelgrinberg** [comment]
@Sphinxs The bug was in Flask-SocketIO. Are you using the latest? Try running `pip install --upgrade flask-socketio`.
